rids/monitors/tls_monitor.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 Jigsaw Operations LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from dataclasses import dataclass
+import ipaddress
+from types import Generator
+from types import Union
+
+from rids.monitors import tshark
+
+
+# union type for IPv4 and IPv6
+IpAddress = Union[ipaddress.IPv4Address, ipaddress.IPv6Address]
+
+@dataclass
+class TlsConnection:
+  """Represents salient properties of a TLS connection and when it was seen."""
+  timestamp: str
+  remote_ip: IpAddress
+  remote_port: int
+  server_name: str
+  ja3: str
+  ja3_full: str
+  ja3s: str
+  ja3s_full: str
+
+@dataclass
+class TlsRule:
```

Comment:
Move this to the TLS matching code (or delete), since it's not needed to get `TlsConnection`s